### PR TITLE
feat: Update es-builtins

### PR DESCRIPTION
**Pull request by scagood**
I took the data from [mdn/browser-compat-data](https://github.com/mdn/browser-compat-data) and used that to update the es-builtins file (it was a lot-a-bit manual).

I do wander if we should make a script to automate this fully.

### CODE REVIEW

File: lib/rules/no-unsupported-features/es-builtins.js [hunk 1]
```diff
@@ -14,136 +14,606 @@ const getConfiguredNodeVersion = require("../../util/get-configured-node-version
...
+            [READ]: { supported: "0.10.0" },
+            dotAll: { [READ]: { supported: "8.10.0" } },
+            hasIndices: { [READ]: { supported: "16.0.0" } },
+            input: { [READ]: { supported: "0.10.0" } },
+            lastIndex: { [READ]: { supported: "0.10.0" } },
+            lastMatch: { [READ]: { supported: "0.10.0" } },
+            lastParen: { [READ]: { supported: "0.10.0" } },
+            leftContext: { [READ]: { supported: "0.10.0" } },
+            n: { [READ]: { supported: "0.10.0" } },
+            rightContext: { [READ]: { supported: "0.10.0" } },
```
Comment (scagood) [inline lib/rules/no-unsupported-features/es-builtins.js:303]: There appear to be some wrong here, I will look more later :thinking:

**Review by scagood — COMMENTED** [review]
This is built using
https://github.com/mdn/browser-compat-data

**Comment by scagood** [comment]
I have a little extension to this in https://github.com/scagood/eslint-plugin-n/pull/1

That removes all the 'backport' properties in favour of just having a supported array.
I will make that PR after this one is either merged or closed :eyes:

**Comment by scagood** [comment]
There is data that we want in both `data.javascript.builtins` and `data.api`.

<details>
<summary>The script I used to help automate this is:</summary>

```js

const data = require("./data.json")

const knownIssues = new Set([
    "TypedArray",
    "AsyncFunction",
    "Generator",
    "AsyncGenerator",
    "GeneratorFunction",
    "AsyncGeneratorFunction",
    "Iterator",
    "AsyncIterator",
    "undefined",
    "AggregateError.errors",
])

function getVersion(data) {
    return (
        data.__compat?.support?.nodejs?.version_added ??
        data.__compat?.support?.nodejs?.[0]?.version_added ??
        false
    )
}

const dataset = data.javascript.builtins

for (const builtin in dataset) {
    if (knownIssues.has(builtin)) {
        continue
    }
    const builtinData =
        builtin.endsWith("Array") && !builtin.startsWith("Array")
            ? {
                  ...dataset.TypedArray,
                  ...dataset[builtin],
              }
            : dataset[builtin]
    const builtinVersion = getVersion(builtinData)

    if (builtinVersion === false) {
        continue
    }

    let lines = [
        `${builtin}: {`,
        `    [READ]: { supported: "${builtinVersion}" },`,
    ]

    for (const fullKey of Object.keys(builtinData)) {
        // if (fullKey.endsWith('_static') === false) {
        //     continue;
        // }

        const key = fullKey //.slice(0, -7);
        const keyData = builtinData[fullKey]
        if (
            key === builtin ||
            key === "__compat" ||
            key.startsWith("@@") ||
            knownIssues.has(`${builtin}.${key}`) ||
            knownIssues.has(`${builtin}.*`) ||
            knownIssues.has(`*.${key}`)
        ) {
            continue
        }

        if (
            // Its a new feature of ${builtin}
            keyData.__compat?.spec_url == null ||
            // Its not a static property
            keyData.__compat?.spec_url.includes(
                `.prototype.${key.toLowerCase()}`
            )
        ) {
            continue
        }

        const keyVersion = getVersion(keyData)
        if (keyVersion === false) {
            continue
        }

        lines.push(`    ${key}: { [READ]: { supported: "${keyVersion}" } },`)
    }

    lines.push("},")

    console.info(lines.join("\n"))
}

```

Even though its super hacky... when the `data.api` dataset is used we need to uncomment the following:

```js
if (fullKey.endsWith('_static') === false) {
    continue;
}

const key = fullKey.slice(0, -7);
```


</details>

**Review by aladdin-add — APPROVED** [review]
LGTM, thanks! 💯

**Pull request merged**